src/core/ready.js

```diff
@@ -49,11 +47,28 @@ jQuery.extend( {
 			return;
 		}
 
-		// If there are functions bound, to execute
-		readyList.resolveWith( document, [ jQuery ] );
+		whenReady = function( fn ) {
+			readyCallbacks.push( fn );
+
+			if ( !++readyFiring ) {
```

Comment:
This is not the easiest to read. Would breaking it up increase the size? What size penalty would it bring to just use booleans?

I'd prefer to such "clever coding" to be performed by Uglify, not in source.